--- a/make_mirror.sh
+++ b/make_mirror.sh
@@ -316,7 +316,11 @@ PROXYPID=$!
 trap 'kill "$PROXYPID" || :' EXIT INT TERM
 
 for i in $(seq 10); do
-	curl --proxy "http://127.0.0.1:8080/" --silent -o /dev/null "http://deb.debian.org/debian/dists/$DEFAULT_DIST/InRelease" && break
+	if env http_proxy="http://127.0.0.1:8080/" wget --quiet -O /dev/null "http://deb.debian.org/debian/dists/$DEFAULT_DIST/InRelease"; then
+		break
+	elif curl --proxy "http://127.0.0.1:8080/" --silent -o /dev/null "http://deb.debian.org/debian/dists/$DEFAULT_DIST/InRelease"; then
+		break
+	fi
 	sleep 1
 done
 if [ ! -s "$newmirrordir/dists/$DEFAULT_DIST/InRelease" ]; then
